### PR TITLE
[217] Develop a Customer details list view page and Save Session in a…

### PR DESCRIPTION
**Pull request by Samju200**
… Cookie

## Task: `Name of given task`
develop1

- `Changes you made to given task`

### CODE REVIEW

File: submissions/adeyemiJuwon/phpDatabase/customers.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
+<?php
+try{
+$pdo = new PDO("mysql:host=localhost;dbname=store", 'root', 'mysql');
+     $pdo->setAttribute(Pdo::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
+    
+ }catch(PDOException $e){
+     die("ERROR:Could not connect." . $e->getMessage());
+ }
+  setcookie("Id",  $row['id'], time()+3600, "/", "", false);
```
Comment (adaugochi) [inline submissions/adeyemiJuwon/phpDatabase/customers.php:9]: Where is `$row` coming from?

**Review by adaugochi — CHANGES_REQUESTED** [review]

File: submissions/adeyemiJuwon/phpDatabase/customers.php (hunk 1 above)
Comment (Samju200) [reply submissions/adeyemiJuwon/phpDatabase/customers.php:9]: Coming from the table from the database. @adaugochi

Comment (Samju200) [reply submissions/adeyemiJuwon/phpDatabase/customers.php]: @adaugochi. I have reformatted my code. help me to review it again. Thanks

File: submissions/adeyemiJuwon/phpDatabase/customers.php [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  
+  if(!isset($_COOKIE['customers'])) {
+  echo "Cookie named '" .'customers' . "' is not set!";
+} else {
+  echo "Cookie '" . 'customers' . "' is set!<br>";
+  echo "Value is: " . $_COOKIE['customers'];
+}
+try{
+    
+    if($results ->rowCount() > 0){
```
Comment (adaugochi) [inline submissions/adeyemiJuwon/phpDatabase/customers.php:12]: The `$results` that you are using here is that from the SQL query. It means you are not using that stored in `cookie` you are always hitting the database

**Review by adaugochi — CHANGES_REQUESTED** [review]
Well done @Samju200 

But, you are still hitting the database each time the page is refresh. So, we want to only hit the database if cookie is not isset. But, if cookie is set, we want to use the data stored in cookie instead,

File: submissions/adeyemiJuwon/phpDatabase/cookies.php [hunk 3]
```diff
@@ -0,0 +1,25 @@
+<?php
+include('connect.php');
+
+         class QueryDatabase {
```
Comment (adaugochi) [inline submissions/adeyemiJuwon/phpDatabase/cookies.php:4]: Format this block of code

File: submissions/adeyemiJuwon/phpDatabase/customers.php [hunk 4]
```diff
@@ -0,0 +1,45 @@
+<?php
+  include('cookies.php');
+  
+  $results = isset($_COOKIE['customers'])? unserialize($_COOKIE['customers']) : $queryDatabase->getAllCustomers();
+
+
+
+if(isset($_COOKIE['customers'])) {
```
Comment (adaugochi) [inline submissions/adeyemiJuwon/phpDatabase/customers.php:8]: There is no need for this here again

**Review by adaugochi — CHANGES_REQUESTED** [review]

File: submissions/adeyemiJuwon/phpDatabase/cookies.php (hunk 3 above)
Comment (Samju200) [reply submissions/adeyemiJuwon/phpDatabase/cookies.php]: Done.
Thanks @adaugochi

File: submissions/adeyemiJuwon/phpDatabase/customers.php (hunk 4 above)
Comment (Samju200) [reply submissions/adeyemiJuwon/phpDatabase/customers.php]: Okay

File: submissions/adeyemiJuwon/phpDatabase/cookies.php [hunk 5]
```diff
@@ -0,0 +1,25 @@
+<?php
+include('connect.php');
+
+         class queryDatabase {
```
Comment (adaugochi) [inline submissions/adeyemiJuwon/phpDatabase/cookies.php:4]: From line 4 below isn't properly formatted

**Review by adaugochi — CHANGES_REQUESTED** [review]

File: submissions/adeyemiJuwon/phpDatabase/cookies.php (hunk 5 above)
Comment (Samju200) [reply submissions/adeyemiJuwon/phpDatabase/cookies.php]: Okay
Done @adaugochi

File: submissions/adeyemiJuwon/phpDatabase/customers.php [hunk 6]
```diff
@@ -0,0 +1,38 @@
+<?php
+  include('cookies.php');
```
Comment (adaugochi) [inline submissions/adeyemiJuwon/phpDatabase/customers.php:2]: Make this look good too

**Review by adaugochi — CHANGES_REQUESTED** [review]

File: submissions/adeyemiJuwon/phpDatabase/connect.php [hunk 7]
```diff
@@ -0,0 +1,11 @@
+<?php 
+  
+      try{
```
[the remark this answers is not in the document]
Comment (Samju200) [reply submissions/adeyemiJuwon/phpDatabase/connect.php]: Okay
Done @adaugochi 
Thanks

**Review by adaugochi — APPROVED** [review]
Well done @Samju200 

But, your solution looks exactly like what Isreal did.

**Comment by Samju200** [comment]
> Well done @Samju200
> 
> But, your solution looks exactly like what Isreal did.

Wow
But i ask him about it, he told he has no solution about it yet then

**Pull request merged**